crates/bevy_core_pipeline/src/overlay/mod.rs

```diff
@@ -0,0 +1,150 @@
+use bevy_app::{App, Plugin};
+use bevy_asset::{load_internal_asset, HandleUntyped};
+use bevy_diagnostic::{Diagnostics, FrameTimeDiagnosticsPlugin};
+use bevy_ecs::{
+    entity::Entity,
+    prelude::With,
+    system::{Commands, Local, Query, Res},
+};
+use bevy_input::{keyboard::KeyCode, Input};
+use bevy_reflect::TypeUuid;
+use bevy_render::{
+    extract_component::ExtractComponentPlugin,
+    prelude::Shader,
+    render_graph::{RenderGraph, SlotInfo, SlotType},
+    renderer::RenderQueue,
+    RenderApp, RenderStage,
+};
+
+mod camera_overlay;
+mod overlay_node;
+mod pipeline;
+use bevy_time::Time;
+use bevy_utils::{Duration, Instant};
+pub use camera_overlay::{CameraOverlay, CameraOverlayBundle};
+
+use crate::overlay::{overlay_node::graph, pipeline::OverlayPipeline};
+
+use self::overlay_node::DiagnosticOverlayBuffer;
+
+pub const OVERLAY_SHADER_HANDLE: HandleUntyped =
+    HandleUntyped::weak_from_u64(Shader::TYPE_UUID, 1236245567947772696);
+
+pub(crate) struct OverlayDiagnostics {
+    avg_fps: f32,
+}
+
+fn extract_overlay_diagnostics(
+    mut commands: Commands,
+    diags: Res<Diagnostics>,
+    time: Res<Time>,
+    mut last_update: Local<Option<Instant>>,
+) {
+    if last_update.is_none() {
+        *last_update = time
+            .last_update()
+            .and_then(|lu| lu.checked_sub(Duration::from_secs(1)));
+    }
+
+    if let Some(last_overlay_update) = *last_update {
+        if let Some(current) = time.last_update() {
+            if (current - last_overlay_update).as_secs_f32() > 0.15 {
+                *last_update = time.last_update();
+                commands.insert_resource(OverlayDiagnostics {
+                    avg_fps: diags
+                        .get(FrameTimeDiagnosticsPlugin::FPS)
+                        .and_then(|diag| diag.average())
+                        .unwrap_or_default() as f32,
+                });
+            }
+        }
+    }
+}
+
+fn prepare_overlay_diagnostics(
+    buffer: Res<DiagnosticOverlayBuffer>,
+    render_queue: Res<RenderQueue>,
+    diagnostics: Res<OverlayDiagnostics>,
+) {
+    if diagnostics.is_changed() {
+        buffer.write_buffer(diagnostics.as_ref(), render_queue.as_ref());
+    }
+}
+
+#[derive(Default)]
+pub struct OverlayPlugin {
+    pub trigger: Option<KeyCode>,
```

Comment:
Presumably this should be a more generic user input, but we don't have that functionality upstream yet.

Reply:
It's sad that the default keybindings get to cheat :(